Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteBufferProxy.cpp

```diff
@@ -63,18 +65,60 @@ static bool offsetOrSizeExceedsBounds(size_t dataSize, WebCore::WebGPU::Size64 o
     return offset >= dataSize || (requestedSize.has_value() && requestedSize.value() + offset > dataSize);
 }
 
-void RemoteBufferProxy::getMappedRange(WebCore::WebGPU::Size64 offset, std::optional<WebCore::WebGPU::Size64> size, Function<void(std::span<uint8_t>)>&& callback)
+WebCore::WebGPU::BufferUsageFlags RemoteBufferProxy::usage() const
 {
-    // FIXME: Implement error handling.
+    return m_usage;
+}
+
+void RemoteBufferProxy::getMappedRange(WebCore::WebGPU::Size64 offset, std::optional<WebCore::WebGPU::Size64> size, Function<void(std::pair<uint64_t, uint64_t>)>&& callback)
+{
+    if (!m_bufferMemory) {
+        callback({ });
+        return;
+    }
+
     auto sendResult = sendSync(Messages::RemoteBuffer::GetMappedRange(offset, size));
-    auto [data] = sendResult.takeReplyOr(std::nullopt);
+    auto [dataOffsetAndSize] = sendResult.takeReplyOr(std::make_pair(0ull, 0ull));
+    auto dataOffset = dataOffsetAndSize.first;
+    auto dataSize = dataOffsetAndSize.second;
 
-    if (!data || !data->data() || offsetOrSizeExceedsBounds(data->size(), offset, size)) {
+    auto data = m_bufferMemory->mutableSpan();
+    if (!data.data() || offsetOrSizeExceedsBounds(data.size(), offset, size)) {
         callback({ });
         return;
     }
 
-    callback(data->mutableSpan().subspan(offset));
+    callback(std::make_pair(dataOffset, dataSize));
+}
+
+std::optional<std::span<uint8_t>> RemoteBufferProxy::mutableSpan()
```

Comment:
What's the difference between a `std::optional<std::span<...>>` and a `std::span<...>` that is empty? Is it meaningful to distinguish between no buffer memory and the buffer being empty?